internal/database/codeowners.go

```diff
@@ -0,0 +1,83 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/database/basestore"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+
+	codeownerspb "github.com/sourcegraph/sourcegraph/internal/own/codeowners/proto"
+)
+
+// CodeownersStore currently allows to store only the most recent version of CODEOWNERS
+// file for main branch of given repository.
+type CodeownersStore interface {
+	basestore.ShareableStore
+	// Put updates the newest CODEOWNERS file for given repository's HEAD.
```

Comment:
Probably should be like this?
```suggestion
	// Put updates of the newest CODEOWNERS file for given repository's HEAD.
```